pipeline_dp/combiners.py

```diff
@@ -392,6 +399,80 @@ def explain_computation(self) -> ExplainComputationReport:
+
+    It can compute any number of quantiles at once with the same accuracy.
+    It uses QuantileTree algorithm from Google C++ DP library
+    https://github.com/google/differential-privacy/blob/main/cc/algorithms/quantile-tree.h
+    with the wrapper from PyDP.
+
+    The accumulator is QuantileTree object serialized to string.
+    """
+
+    AccumulatorType = str
+
+    def __init__(self, params, percentiles: List[float]):
+        self._params = params
+        self._percentiles = percentiles
+        self._quantiles = [p / 100 for p in percentiles]
+
+    def create_accumulator(self, values) -> AccumulatorType:
+        tree = self._create_empty_quantile_tree()
+        for value in values:
+            tree.add_entry(value)
+        return tree.serialize().to_bytes()
+
+    def merge_accumulators(self, accumulator1: AccumulatorType,
+                           accumulator2: AccumulatorType) -> AccumulatorType:
+        tree = self._create_empty_quantile_tree()
+
+        tree.merge(pydp._pydp.bytes_to_summary(accumulator1))
+        tree.merge(pydp._pydp.bytes_to_summary(accumulator2))
+        return tree.serialize().to_bytes()
+
+    def compute_metrics(self, accumulator: AccumulatorType) -> AccumulatorType:
+        tree = self._create_empty_quantile_tree()
+        tree.merge(pydp._pydp.bytes_to_summary(accumulator))
+
+        quantiles = tree.compute_quantiles(
+            self._params.eps, self._params.delta,
+            self._params.aggregate_params.max_partitions_contributed,
+            self._params.aggregate_params.max_contributions_per_partition,
+            self._quantiles, self._noise_type())
+
+        return dict([(name, value)
+                     for name, value in zip(self.metrics_names(), quantiles)])
+
+    def metrics_names(self) -> List[str]:
+
+        def format_metric_name(p: float):
+            int_p = int(round(p))
+            if int_p == p:
+                p = int_p
+            else:
+                p = str(p).replace('.', '_')
+            return f"percentile_{p}"
+
+        return list(map(format_metric_name, self._percentiles))
+
+    def explain_computation(self) -> ExplainComputationReport:
+        return lambda: f"Computed percentiles {self._percentiles} with (eps={self._params.eps} delta={self._params.delta})"
+
+    def _create_empty_quantile_tree(self):
+        # The default tree parameters taken from
+        # https://github.com/google/differential-privacy/blob/605ec87bcbd4a536995b611132dbf4d341d2e91d/cc/algorithms/quantile-tree.h#L47
+        DEFAULT_TREE_HEIGHT = 4
+        DEFAULT_BRANCHING_FACTOR = 16
+        return quantile_tree.QuantileTree(
+            self._params.aggregate_params.min_value,
+            self._params.aggregate_params.max_value, DEFAULT_TREE_HEIGHT,
+            DEFAULT_BRANCHING_FACTOR)
+
+    def _noise_type(self) -> str:
+        return "laplace" if self._params.aggregate_params.noise_kind == pipeline_dp.NoiseKind.LAPLACE else "gaussian"
```

Comment:
This assumes there are no other noise types in future.
Can we use something more stable, e.g. 
`return self._params.aggregate_params.noise_kind.value`?